src/java/voldemort/utils/MetadataVersionStoreUtils.java

```diff
@@ -80,7 +80,12 @@ public static Properties mergeVersions(Properties prop1, Properties prop2) {
             return prop1;
         }
 
-        Properties result = new Properties(prop1);
```

Comment:
Why not use Voldemort's `Props` class instead of Java `Properties`? Voldemort's implementation already supports this merging use case. Anyway, no need to do any major code overhaul as part of this PR, but it may be worth at least marking this `mergeVersions` function as `@Deprecated` since we should probably avoid using Java `Properties`.